File: 28062024/restFinalFedericoRifugiato/src/main/java/com/federicorifugiato/model/Ruolo.java
package com.federicorifugiato.model;

import java.util.ArrayList;
import java.util.List;

import com.federicorifugiato.enums.TipoRuolo;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;

@Entity
public class Ruolo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)    
    @Column(name = "ID_G")
    private int id;
    
    @Enumerated(EnumType.STRING)    
    @Column(name = "TIPOLOGIA")
    private TipoRuolo tipologia;
    
    @ManyToMany(mappedBy = "ruoli", cascade = CascadeType.REFRESH, fetch = FetchType.EAGER)
    private List<Utente> utenti = new ArrayList<>();

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public TipoRuolo getTipologia() {
		return tipologia;
	}

	public void setTipologia(TipoRuolo tipologia) {
		this.tipologia = tipologia;
	}

	public List<Utente> getUtenti() {
		return utenti;
	}

	public void setUtenti(List<Utente> utenti) {
		this.utenti = utenti;
	}
    
}

File: 05072024/FrontEnd/src/app/dto/Ruolo.ts
import { TipoRuolo } from '../enum/TipoRuolo';

export class Ruolo {
  private _id: number;
  private _tipologia: TipoRuolo;

  constructor(id: number, tipologia: TipoRuolo) {
    this._id = id;
    this._tipologia = tipologia;
  }

  public get id(): number {
    return this._id;
  }

  public set id(id: number) {
    this._id = id;
  }

  public get tipologia(): TipoRuolo {
    return this._tipologia;
  }

  public set tipologia(tipologia: TipoRuolo) {
    this._tipologia = tipologia;
  }
}
